src/services/createRouter.ts

```diff
@@ -123,7 +126,18 @@ export function createRouter<const TRoutes extends Routes, const TOptions extend
         throw new Error(`Switch is not exhaustive for before hook response status: ${JSON.stringify(beforeResponse satisfies never)}`)
     }
 
-    propStore.setProps(to)
+    try {
+      propStore.setProps(to)
+    } catch (error) {
+      if (error instanceof RouterPushError) {
+        await push(...error.to as Parameters<RouterPush>)
```

Comment:
Curious why we need the `as` here. We have something similar for hooks but no `as`.

Reply:
we do the casting [here](https://github.com/kitbagjs/router/blob/main/src/services/hooks.ts#L90)